tests/rptest/tests/limits_test.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2020 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import random
+
+from rptest.clients.default import DefaultClient
+from rptest.clients.rpk import RpkException, RpkTool
+from rptest.services import redpanda
+from rptest.services.admin import Admin
+from rptest.services.cluster import cluster
+
+from rptest.clients.types import TopicSpec
+from rptest.tests.redpanda_test import RedpandaTest
+import string
+
+from rptest.util import wait_until
+
+
+class LimitsTest(RedpandaTest):
+    def _rand_string(sz):
+        return ''.join(
+            random.choice(string.ascii_letters + string.digits)
+            for _ in range(sz))
+
+    @cluster(num_nodes=3)
+    def test_kafka_request_max_size(self):
+        topic = TopicSpec(partition_count=1)
+        # create topic
+        DefaultClient(self.redpanda).create_topic(topic)
+        rpk = RpkTool(self.redpanda)
+        # produce 512KB message
+        rpk.produce(topic=topic.name,
+                    key="test",
+                    msg=LimitsTest._rand_string(512 * 1024))
+        # set request max bytes to 50KB
+        new_limit = 50 * 1024
+        rpk.cluster_config_set("kafka_request_max_bytes", str(new_limit))
+
+        admin = Admin(self.redpanda)
+
+        def configuration_propagated():
+            values = [
+                admin.get_cluster_config(node=n)["kafka_request_max_bytes"]
+                for n in self.redpanda.nodes
+            ]
+
+            return all([v == new_limit for v in values])
+
+        wait_until(configuration_propagated, 10, 0.2)
+        # try producing once again, it should fail as 512 KB is larger than a limit
+        failed = False
+        try:
+            rpk.produce(topic=topic.name,
+                        key="test",
+                        msg=LimitsTest._rand_string(512 * 1024))
+        except RpkException as e:
```

Comment:
nit: instead of waiting until the configuration is propagated to the cluster and then producing, produce _until_ the expected failure occurs. that is a more direct representation of what happens in a real-world context.